system/lib/libc/emscripten_memcpy.c

```diff
@@ -29,7 +29,7 @@ static void *__memcpy(void *restrict dest, const void *restrict src, size_t n) {
   unsigned char *block_aligned_d_end;
   unsigned char *d_end;
 
-#ifndef EMSCRIPTEN_STANDALONE_WASM
+#if !defined(EMSCRIPTEN_STANDALONE_WASM) || defined(__wasm_bulk_memory__)
```

Comment:
Perhaps it's worth replacing all of `memcpy` if bulk memory is enabled? It can all be a single bulk memory operation in that case, which is smaller. Unless we have data showing it's less fast on small sizes or something like that?

Reply:
Perhaps we can do that as a followup..  this change is nice and compact because it simply replaces the exists `_big` JS functions with a native function.   Collecting data on whether to shrink or remove the threshold seems like it can be separate thing we can look into after this lands.

Reply:
Ok, sounds good. Since there is a speed benefit here, it seems worth a temporary size tradeoff for it. But I am very optimistic that replacing all of memcpy will end up better later.

Reply:
Yes, once bulk memory is the default we can/should just always use it here, and depend on binaryen to lower it away.  Until it that time though we don't want to depend on binaryen in the default case.

Reply:
Ah, right, makes sense to wait for bulk memory to be on by default.